.github/workflows/check_lint_build_release.yaml

```diff
@@ -27,189 +31,102 @@ jobs:
         uses: bufbuild/buf-action@v1
         with:
           setup_only: true
-      
+
       # Disabled, since buf format is not configurable
       #- name: Buf format
       #  run: buf format -d --exit-code
-      
+
       - name: Buf lint
         run: buf lint --error-format github-actions
 
-      - name: Install latest stable toolchain
-        uses: dtolnay/rust-toolchain@master
-        with:
-          toolchain: stable
-          components: rustfmt, clippy
-
-      - name: Rust Cache
-        uses: Swatinem/rust-cache@v2.5.1
-        with:
-          prefix-key: "ubuntu-stable"
-
-      - name: Rustfmt
-        run: cargo fmt --all -- --check
-      
       - name: Cargo check
         run: cargo check
-      
+
       - name: Clippy
         run: cargo clippy --all-targets --all-features
-      
-      - name: Build
-        run: cargo build --release
-
-      - name: 'Set environment variables: version number and output filenames'
-        run: |
-          APP_VERSION=$(cargo metadata --format-version 1 | \
-            jq -er '.packages | map(select(.name == "bip300301_enforcer") | .version) | .[0]')
-          APP_FILENAME="bip300301_enforcer-${APP_VERSION}-x86_64-unknown-linux-gnu"
-          echo "APP_VERSION=$APP_VERSION" >> "$GITHUB_ENV"
-          echo "APP_FILENAME=$APP_FILENAME" >> "$GITHUB_ENV"
-
-      - name: 'Set filenames for release binaries'
-        run: |
-          pushd "target/release"
-          ln -fs "bip300301_enforcer" "${APP_FILENAME}"
-          popd
 
-      - name: 'Upload artifacts'
-        uses: actions/upload-artifact@v4
-        with:
-          name: ${{ env.APP_FILENAME }}
-          if-no-files-found: error
-          path: |
-            target/release/${{ env.APP_FILENAME }}
-
-      - name: Release
-        uses: softprops/action-gh-release@v2.0.2
-        if: startsWith(github.ref, 'refs/tags/')
-        with:
-          files: |
-            target/release/${{ env.APP_FILENAME }}
-          fail_on_unmatched_files: true
-
-  build-release-macos:
-    name: Build, Release (macos-x86_64)
-    # macos-latest is ARM64, macos-latest-large is x86-64
-    runs-on: macos-latest-large
+  build-release:
+    strategy:
+      fail-fast: false
+      matrix:
+        include:
+          - os: ubuntu-latest
+            name: x86_64-unknown-linux-gnu
+            binary-suffix: ""
+            rustflags: ""
+
+          - os: macos-latest-large
+            name: x86_64-apple-darwin
+            binary-suffix: ""
+            rustflags: ""
+
+          - os: ubuntu-latest
+            name: x86_64-pc-windows-gnu
+            binary-suffix: .exe
+            rustflags: "-C linker=/usr/bin/x86_64-w64-mingw32-gcc"
+
+    name: Check, Lint, Build, Release (${{ matrix.name }})
```

Comment:
This should be named eg. `Build, Release (${{ matrix.name }})`, if the check/lint step is skipped